### PR TITLE
Feature/option prepend username

### PR DESCRIPTION
**Pull request by junhuicoding**
Features added based on https://github.com/austinhuang0131/barinsta/issues/631:
- Added a user setting for pretending username to downloaded filenames.
image
When turned on (Tested on albums, stories, video and image posts):
image
When turned off:
image

- The current language for the setting is only in English `Prepend Username to Filename`, will need to be translated in the respective `strings.xml` files

- ~~Added a null check for `profileModel` in `ProfileFragment.fetchProfileDetails()` as I was running into issues displaying the profile tab when I re-ran the app from android studios when logged in previously. App will just crash. Might want to look into it a bit further, but this null check should not be significant, it just prevents a crash.~~

Additional comments:
This is my first time working on an android app, any suggestions, comments or improvements to be made are extremely welcomed!

### CODE REVIEW

**Comment by junhuicoding** [comment]
ProfileFragment.fetchProfileDetails() has been fixed in master.

File: app/src/main/java/awais/instagrabber/fragments/settings/DownloadsPreferencesFragment.java [hunk 1]
```diff
@@ -49,6 +50,14 @@ private Preference getSaveToCustomFolderPreference(@NonNull final Context contex
                 .show(getParentFragmentManager(), null));
     }
 
+    private Preference getPrependUsernameToFilenamePreference(@NonNull final Context context) {
+        final SwitchPreferenceCompat preference = new SwitchPreferenceCompat(context);
+        preference.setKey(Constants.DOWNLOAD_PREPEND_USER_NAME);
+        preference.setTitle("Prepend Username to Filename");
```
Comment (austinhuang0131) [inline app/src/main/java/awais/instagrabber/fragments/settings/DownloadsPreferencesFragment.java:56]:
```suggestion
        preference.setTitle(R.string.download_prepend_username);
```

File: app/src/main/java/awais/instagrabber/utils/DownloadUtils.java [hunk 2]
```diff
@@ -116,23 +116,34 @@ private static void dmDownloadImpl(@NonNull final Context context,
...
 
     @NonNull
     private static File getDownloadSaveFile(final File finalDir,
                                             final String postId,
                                             final String sliderPostfix,
-                                            final String displayUrl) {
-        final String fileName = postId + sliderPostfix + getFileExtensionFromUrl(displayUrl);
+                                            final String displayUrl,
+                                            final String username) {
+        final String usernamePrepend = "".equals(username) ? "" : "@" + username + "_";
```
Comment (austinhuang0131) [inline app/src/main/java/awais/instagrabber/utils/DownloadUtils.java:145]:
```suggestion
        final String usernamePrepend = TextUtils.isEmpty(username) ? "" : "@" + username + "_";
```

**Review by austinhuang0131 — CHANGES_REQUESTED** [review]
Thank you for your contributions.

One thing you missed is that Barinsta does depend on file name to check if a media (only for posts, not stories and stuff) is downloaded or not, through `DownloadUtils$checkDownloaded`. You want to make sure it successfully checks for the downloaded status.

File: app/src/main/java/awais/instagrabber/fragments/settings/DownloadsPreferencesFragment.java (hunk 1 above)
Comment (junhuicoding) [reply app/src/main/java/awais/instagrabber/fragments/settings/DownloadsPreferencesFragment.java]: Fixed!

File: app/src/main/java/awais/instagrabber/utils/DownloadUtils.java (hunk 2 above)
Comment (junhuicoding) [reply app/src/main/java/awais/instagrabber/utils/DownloadUtils.java]: Fixed!

**Comment by junhuicoding** [comment]
> Thank you for your contributions.
> 
> One thing you missed is that Barinsta does depend on file name to check if a media (only for posts, not stories and stuff) is downloaded or not, through `DownloadUtils$checkDownloaded`. You want to make sure it successfully checks for the downloaded status.

Thanks for the feedback! I've added your suggested changes, as well as included the `usernamePrepend` string to `getDownloadChildSaveFile` correctly for `MEDIA_TYPE_SLIDER`.  `MEDIA_TYPE_IMAGE` and  `MEDIA_TYPE_VIDEO` were already added.

This will not flag a downloaded item correctly if a user changes the setting to prepend username after downloading some files due to a mismatch in names. This is the same as the other download settings related to changing of folders. This should not have a large impact as users are unlikely to switch between these settings regularly. As of now, the feature works in correctly displaying downloaded icon for files downloaded when the settings at time of download are the same as the current settings.

- One potential fix for the prepending option will be to search for the existent of both filenames, with and without username prepend (simple to implement)
- Checking for the username specific folder will possible to implement too, just check for both the username folder as well as general folder
- The issue for a custom folder would be tricky, as there is no history of all previously selected folders. It might be possible to check for all folders that barinsta has permission to write into, but I am not too sure about this route.

**Comment by austinhuang0131** [comment]
Oh, you're thinking too much... We're just talking current folder and assume it's not renamed

**Comment by junhuicoding** [comment]
> Oh, you're thinking too much... We're just talking current folder and assume it's not renamed
> 
> So, yes, check with and without username, c'est tout (and username-specific folder as well as current custom folder is already implemented within)

Thanks for the clarifications! I've updated the changes requested for checking with and without username.

**Review by austinhuang0131 — APPROVED** [review]

**Pull request merged**